### DISCUSSION TITLE
recommended way to add recurring logic

### DISCUSSION THREAD

**Discussion by aruder77** [microhomie/microhomie#57] [Q&A]
Hi,

my homie nodes must execute logic which is not triggert by mqtt events but in a recurring way, e.g. every 100ms.
What is the best approach to do that? The main control loop is delegated to microhomie...

Thanks,
Axel

**Comment by rroemhild** [comment]
Hi Axel,

Microhomie ist based on the MicroPython asyncio feature. The best way would be to run your code in a coroutine.

The _dht_ and the _pir_ examples uses coroutines. The _dht_ example simply starts a coroutine for temperature and humidity measurement and writes the results to the properties, while the _pir_ example has the extra feature to start and stop the coroutine over mqtt.

Best regards,
Rafael

**Reply by aruder77** [reply]
Hi Rafael,
great! Thanks a lot, that is exactly what I was looking for. I first didn't recognise the solution in the examples, thanks for pointing me that way.

Thanks,
Axel
